crates/optimism/payload/src/builder.rs

```diff
@@ -460,26 +459,6 @@ where
     // create the block header
     let transactions_root = proofs::calculate_transaction_root(&executed_txs);
 
-    // initialize empty blob sidecars. There are no blob transactions on L2.
-    let blob_sidecars = Vec::new();
-    let mut excess_blob_gas = None;
-    let mut blob_gas_used = None;
-
-    // only determine cancun fields when active
-    if chain_spec.is_cancun_active_at_timestamp(attributes.payload_attributes.timestamp) {
-        excess_blob_gas = if chain_spec.is_cancun_active_at_timestamp(parent_block.timestamp) {
-            let parent_excess_blob_gas = parent_block.excess_blob_gas.unwrap_or_default();
-            let parent_blob_gas_used = parent_block.blob_gas_used.unwrap_or_default();
-            Some(calc_excess_blob_gas(parent_excess_blob_gas, parent_blob_gas_used))
-        } else {
-            // for the first post-fork block, both parent.blob_gas_used and
-            // parent.excess_blob_gas are evaluated as 0
-            Some(calc_excess_blob_gas(0, 0))
-        };
-
-        blob_gas_used = Some(0);
-    }
```

Comment:
we can simplify this yes, but the 4844 fields must still be none before cancun

Reply:
Ah, do you mean to support people who build blocks pre-Cancun/Etocone per custom hardfork rules or past timestamps? We dodge that as we start from at least Granite but core Reth should be general for sure. What do you think of setting `(Some(0), Some(0))` or `(None, None)` based on the `is_cancun_active_at_timestamp` check?

Reply:
this API is also used by OP to resync the chain from genesis for example, which basically sends every block as a list of txs via payload attributes so this must follow all fork rules

and yes either (Some(0), Some(0)) or none

Reply:
> this API is also used by OP to resync the chain from genesis

Interesting, I didn't know this. One last question but should it be Etocone or it's actually Cancun?

Reply:
Thanks, I've updated accordingly :pray:.